marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/getUris.sjs

```diff
@@ -0,0 +1,22 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+const core = require('/data-hub/5/artifacts/core.sjs')
+
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-mapping", "execute");
+
+var stepName, limit;
+let mappingStep = core.getArtifact("mapping", stepName);
```

Comment:
Same thing here, I think this needs to be more defensive. If sourceQuery doesn't exist, return an empty array. If the step doesn't exist, throw an error. If the sourceDatabase doesn't exist, don't do an eval.